cmd/createorupdate/createorupdate.go

```diff
@@ -441,6 +449,146 @@ func createResourceGroup(log *logrus.Entry) error {
 	return err
 }
 
+func gatherArtifacts(artifactDir, artifactKubeconfig string) error {
+	config, err := clientcmd.BuildConfigFromFlags("", artifactKubeconfig)
+	if err != nil {
+		return err
+	}
+	kc, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+
+	// gather node info
+	if err := gatherNodes(kc, artifactDir); err != nil {
+		return err
+	}
+
+	// gather pods from all namespaces
+	// TODO: Ensure we don't leak any secrets. Fix either one of the following:
+	// https://github.com/openshift/openshift-azure/issues/567
+	// https://github.com/openshift/openshift-azure/issues/687
+	// if err := gatherPods(kc, artifactDir); err != nil {
+	//	return err
+	// }
+
+	// gather events from all namespaces
+	if err := gatherEvents(kc, artifactDir); err != nil {
+		return err
+	}
+
+	// gather control plane logs
+	ns := "kube-system"
+	if err := gatherLogs(kc, artifactDir, ns, "sync-master-000000"); err != nil {
+		return err
+	}
+	// TODO: Get logs from the api server and etcd dynamically by using the master count
+	if err := gatherLogs(kc, artifactDir, ns, "master-etcd-master-000000"); err != nil {
```

Comment:
I think at some point we need to move pod names to constants, as they starting to appear all around the code

Reply:
There is a TODO just above about constructing these names dynamically.